llvm/lib/Target/ARM/ARMPredicates.td

```diff
@@ -228,6 +228,10 @@ def DontGenExecuteOnly : Predicate<"!Subtarget->genExecuteOnly()">;
 def GenT1ExecuteOnly : Predicate<"Subtarget->genExecuteOnly() && "
                                  "Subtarget->isThumb1Only() && "
                                  "!Subtarget->hasV8MBaselineOps()">;
+let RecomputePerFunction = 1 in {
+  def TailCallNotR12 : Predicate<[{ MF->getInfo<ARMFunctionInfo>()->shouldSignReturnAddress(true) }]>;
+  def TailCall : Predicate<[{ !MF->getInfo<ARMFunctionInfo>()->shouldSignReturnAddress(true) }]>;
```

Comment:
This name isn't very descriptive, it doesn't affect the presence of tail-calls, just one detail of them. How about `TailCallAnyReg`, or describe the condition instead of the effect, like `SignRetAddr` and `NoSignRetAddr`?

Reply:
OK, thanks. I've changed the name.